numpy/random/bit_generator.pyx

```diff
@@ -43,12 +43,8 @@ except ImportError:
     from random import SystemRandom
     randbits = SystemRandom().getrandbits
 
-try:
-    from threading import Lock
-except ImportError:
-    from dummy_threading import Lock
-
 from cpython.pycapsule cimport PyCapsule_New
+from fastrlock.rlock cimport create_fastrlock as Lock
```

Comment:
This is a new dependency.

Reply:
I agree that we shouldn't add this as a dependency. i just needed a quick way to show that the threading module from python was likely bloated.

An alternative is to do:
```
from _thread import acquire_lock as Lock
```
but there you are importing from private CPython, which is likely incompatible with other implementations of python.

Reply:
yeah that's not a good idea. can you think of a robust alternative? if not, better to just revert

Reply:
I think we can use:
```
from cpython cimport pythread
self.lock = pythread.PyThread_allocate_lock()
```
https://github.com/scoder/fastrlock/blob/master/fastrlock/rlock.pyx#L21